diff --git a/configure.ac b/configure.ac
index 48bc1c5..28c9db0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -55,6 +55,12 @@ AC_SUBST(DAMAGEEXT_VERSION)
 # Obtain compiler/linker options for dependencies
 PKG_CHECK_MODULES(XDAMAGE, [damageproto >= $DAMAGEEXT_VERSION xfixes fixesproto xextproto x11 xproto >= 7.0.22])
 
+if [[ "x${target}" != "x" ]] ; then
+    CFLAGS="$CFLAGS $(echo $CFLAGS_FOR_BUILD | sed -e 's/\/host\//\/target\/usr\//g') --sysroot=$STAGING_DIR"
+    CXXFLAGS="$CXXFLAGS $(echo $CXXFLAGS_FOR_BUILD | sed -e 's/\/host\//\/target\/usr\//g') --sysroot=$STAGING_DIR"
+    LDFLAGS="$LDFLAGS $(echo $LDFLAGS_FOR_BUILD | sed -e 's/\/host\//\/target\/usr\//g') --sysroot=$STAGING_DIR"
+fi
+
 AC_CONFIG_FILES([Makefile
 		src/Makefile
 		xdamage.pc])
diff --git a/ltmain.sh b/ltmain.sh
index 2a50d7f..8cdeb99 100644
--- a/ltmain.sh
+++ b/ltmain.sh
@@ -2458,6 +2458,7 @@ func_enable_tag ()
 # release of libtool.
 func_check_version_match ()
 {
+    macro_revision=2.4.7
     if test "$package_revision" != "$macro_revision"; then
       if test "$VERSION" != "$macro_version"; then
         if test -z "$macro_version"; then
