diff --git a/configure.ac b/configure.ac
index efd5638..4d35b74 100644
--- a/configure.ac
+++ b/configure.ac
@@ -111,6 +111,12 @@ AC_SUBST(xorgconfdir)
 DRIVER_NAME=libinput
 AC_SUBST([DRIVER_NAME])
 
+if [[ "x${target}" != "x" ]] ; then
+    CFLAGS="$CFLAGS $(echo $CFLAGS_FOR_BUILD | sed -e 's/\/host\//\/target\/usr\//g') --sysroot=$STAGING_DIR"
+    CXXFLAGS="$CXXFLAGS $(echo $CXXFLAGS_FOR_BUILD | sed -e 's/\/host\//\/target\/usr\//g') --sysroot=$STAGING_DIR"
+    LDFLAGS="$LDFLAGS $(echo $LDFLAGS_FOR_BUILD | sed -e 's/\/host\//\/target\/usr\//g') --sysroot=$STAGING_DIR"
+fi
+
 AC_CONFIG_FILES([Makefile
 		 include/Makefile
 		 src/Makefile
