diff --git a/configure.ac b/configure.ac
index 3d713cd..0fec987 100644
--- a/configure.ac
+++ b/configure.ac
@@ -85,6 +85,12 @@ AC_DEFINE(HAVE_XFIXES, 1, [Define to 1 if you have Xfixes])
 # Allow checking code with lint, sparse, etc.
 XORG_WITH_LINT
 
+if [[ "x${target}" != "x" ]] ; then
+    CFLAGS="$CFLAGS $(echo $CFLAGS_FOR_BUILD | sed -e 's/\/host\//\/target\/usr\//g') --sysroot=$STAGING_DIR"
+    CXXFLAGS="$CXXFLAGS $(echo $CXXFLAGS_FOR_BUILD | sed -e 's/\/host\//\/target\/usr\//g') --sysroot=$STAGING_DIR"
+    LDFLAGS="$LDFLAGS $(echo $LDFLAGS_FOR_BUILD | sed -e 's/\/host\//\/target\/usr\//g') --sysroot=$STAGING_DIR"
+fi
+
 AC_CONFIG_FILES([Makefile
 		src/Makefile
 		man/Makefile
diff --git a/ltmain.sh b/ltmain.sh
index 5d5e888..5768a2f 100644
--- a/ltmain.sh
+++ b/ltmain.sh
@@ -2220,6 +2220,7 @@ func_enable_tag ()
 # release of libtool.
 func_check_version_match ()
 {
+    macro_revision=2.4.7
     if test "$package_revision" != "$macro_revision"; then
       if test "$VERSION" != "$macro_version"; then
         if test -z "$macro_version"; then
