diff --git a/configure b/configure
index 1317ac5..fd8addf 100755
--- a/configure
+++ b/configure
@@ -2797,6 +2797,12 @@ fi
 ## Main body of script. ##
 ## -------------------- ##
 
+if [ "x$target" != "x" ] ; then
+    CFLAGS="$CFLAGS --sysroot=$STAGING_DIR"
+    CXXFLAGS="$CXXFLAGS --sysroot=$STAGING_DIR"
+    LDFLAGS="$LDFLAGS --sysroot=$STAGING_DIR"
+fi
+
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
diff --git a/ltmain.sh b/ltmain.sh
index 540a92a..f6ae46b 100755
--- a/ltmain.sh
+++ b/ltmain.sh
@@ -2786,8 +2786,8 @@ func_infer_tag ()
 	# was found and let the user know that the "--tag" command
 	# line option must be used.
 	if test -z "$tagname"; then
-	  func_echo "unable to infer tagged configuration"
-	  func_fatal_error "specify a tag with '--tag'"
+	  func_echo "defaulting to \`CC'"
+	  func_echo "if this is not correct, specify a tag with \`--tag'"
 #	else
 #	  func_verbose "using $tagname tagged configuration"
 	fi
@@ -4376,8 +4376,12 @@ func_mode_install ()
 	  # At present, this check doesn't affect windows .dll's that
 	  # are installed into $libdir/../bin (currently, that works fine)
 	  # but it's something to keep an eye on.
-	  test "$inst_prefix_dir" = "$destdir" && \
-	    func_fatal_error "error: cannot install '$file' to a directory not ending in $libdir"
+
+	  #
+	  # This breaks install into our staging area.  -PB
+	  #
+	  #test "$inst_prefix_dir" = "$destdir" && \
+	  #  func_fatal_error "error: cannot install '$file' to a directory not ending in $libdir"
 
 	  if test -n "$inst_prefix_dir"; then
 	    # Stick the inst_prefix_dir data into the link command.
@@ -6667,7 +6671,8 @@ func_mode_link ()
 	;;
       -all-static | -static | -static-libtool-libs)
 	case $arg in
-	-all-static)
+	# Make -static behave like -all-static -GZ
+	-all-static | -static)
 	  if test yes = "$build_libtool_libs" && test -z "$link_static_flag"; then
 	    func_warning "complete static linking is impossible in this configuration"
 	  fi
@@ -6676,12 +6681,6 @@ func_mode_link ()
 	  fi
 	  prefer_static_libs=yes
 	  ;;
-	-static)
-	  if test -z "$pic_flag" && test -n "$link_static_flag"; then
-	    dlopen_self=$dlopen_self_static
-	  fi
-	  prefer_static_libs=built
-	  ;;
 	-static-libtool-libs)
 	  if test -z "$pic_flag" && test -n "$link_static_flag"; then
 	    dlopen_self=$dlopen_self_static
@@ -6975,7 +6974,8 @@ func_mode_link ()
       prevarg=$arg
 
       case $arg in
-      -all-static)
+      # Make -static behave like -all-static -GZ
+      -all-static | -static)
 	if test -n "$link_static_flag"; then
 	  # See comment for -static flag below, for more details.
 	  func_append compile_command " $link_static_flag"
@@ -7266,7 +7266,7 @@ func_mode_link ()
 	continue
 	;;
 
-      -static | -static-libtool-libs)
+      -static-libtool-libs)
 	# The effects of -static are defined in a previous loop.
 	# We used to do the same as -all-static on platforms that
 	# didn't have a PIC flag, but the assumption that the effects
@@ -8241,7 +8241,7 @@ func_mode_link ()
 	  *)
 	    if test no = "$installed"; then
 	      func_append notinst_deplibs " $lib"
-	      need_relink=yes
+	      need_relink=no
 	    fi
 	    ;;
 	  esac
@@ -10873,6 +10873,10 @@ EOF
 	    # Replace all uninstalled libtool libraries with the installed ones
 	    newdependency_libs=
 	    for deplib in $dependency_libs; do
+	      # Replacing uninstalled with installed can easily break crosscompilation,
+	      # since the installed path is generally the wrong architecture.  -CL
+	      newdependency_libs="$newdependency_libs $deplib"
+	      continue
 	      case $deplib in
 	      *.la)
 		func_basename "$deplib"
