diff --git a/configure.ac b/configure.ac
index 2d8870d..1388d24 100644
--- a/configure.ac
+++ b/configure.ac
@@ -78,6 +78,12 @@ AC_SUBST([sdkdir])
 DRIVER_NAME=evdev
 AC_SUBST([DRIVER_NAME])
 
+if [[ "x${target}" != "x" ]] ; then
+    CFLAGS="$CFLAGS $(echo $CFLAGS_FOR_BUILD | sed -e 's/\/host\//\/target\/usr\//g') --sysroot=$STAGING_DIR"
+    CXXFLAGS="$CXXFLAGS $(echo $CXXFLAGS_FOR_BUILD | sed -e 's/\/host\//\/target\/usr\//g') --sysroot=$STAGING_DIR"
+    LDFLAGS="$LDFLAGS $(echo $LDFLAGS_FOR_BUILD | sed -e 's/\/host\//\/target\/usr\//g') --sysroot=$STAGING_DIR"
+fi
+
 AC_CONFIG_FILES([Makefile
                  src/Makefile
                  man/Makefile
